pkg/kubelet/im/image_gc.go

```diff
@@ -104,38 +97,57 @@ func newImageManager(runtime container.Runtime, cadvisorInterface cadvisor.Inter
 	if policy.LowThresholdPercent > policy.HighThresholdPercent {
 		return nil, fmt.Errorf("LowThresholdPercent %d can not be higher than HighThresholdPercent %d", policy.LowThresholdPercent, policy.HighThresholdPercent)
 	}
-	im := &realImageManager{
+
+	im := &realImageGC{
 		runtime:      runtime,
 		policy:       policy,
+		minAge:       defaultGCAge,
 		imageRecords: make(map[string]*imageRecord),
 		cadvisor:     cadvisorInterface,
 		recorder:     recorder,
 		nodeRef:      nodeRef,
-		initialized:  false,
 	}
 
 	return im, nil
 }
 
-func (im *realImageManager) Start() error {
-	go util.Until(func() {
-		// Initial detection make detected time "unknown" in the past.
-		var ts time.Time
-		if im.initialized {
-			ts = time.Now()
-		}
-		err := im.detectImages(ts)
```

Comment:
So [this line](https://github.com/kubernetes/kubernetes/pull/18151/files#diff-632dbe0678962cb022748a3dfdff09d8R116) will not give the usage of images alone. 
`freeSpace` isn't invoked unless there is disk pressure. See [here](https://github.com/mqliang/kubernetes/blob/imageGC/pkg/kubelet/im/image_gc.go#L132). 
We need to update internal state often to keep the image usage information up to date.